src/Auth0Client/Auth0Client.cs

```diff
@@ -303,32 +302,31 @@ private static bool ScopeHasOfflineAccess(string scope)
 				.Any(e => e.Equals("offline_access", StringComparison.InvariantCultureIgnoreCase));
 		}
 
-		protected void SetupCurrentUser(IDictionary<string, string> accountProperties)
+		protected async Task SetupCurrentUser(IDictionary<string, string> accountProperties)
 		{
             var endpoint = string.Format(Auth0Constants.UserInfoEndpoint, this.Domain, accountProperties["access_token"]);
 
 			var request = new Request ("GET", new Uri(endpoint));
-			request.GetResponseAsync ().ContinueWith (t => 
+			var response = await request.GetResponseAsync();
```

Comment:
Done. Some of the methods should be renamed per .NET guidelines... more comments inc